src/main/java/com/ocelotslovebirds/birdhaus/mobai/HangAroundBirdhouseGoal.java

```diff
@@ -31,11 +38,53 @@ public HangAroundBirdhouseGoal(PathfinderMob pMob, double pSpeedModifier,
      * "RandomWanderGoal".
      */
     @Override
-       protected Vec3 getPosition() {
+    protected Vec3 getPosition() {
         if (this.bHousePosVec != null) {
+            this.maybeCleanBhouse();
             return this.bHousePosVec;
         } else {
             return super.getPosition();
         }
     }
+
+    /**
+     * Checks if the birdhouse linked to the parrot still exists, if it doesn't, it
+     * calls the routine used to clear the birdhouses position.
+     */
+    private void maybeCleanBhouse() {
+        if (this.verifyBHouseTicker.tick()) {
+            if (!this.bhouseStillExists()) {
+                this.forgetBirdHouse();
+            }
+        }
+    }
+
+    /**
+     * @return returns true if the birdhouse the parrot is following is still there, false otherwise
+     */
+    private boolean bhouseStillExists() {
+        if (this.mob.level.getBlockState(this.bHouseBlockPos).getBlock() instanceof BirdhouseBlock) {
+            return true;
```

Comment:
I didnt get a chance to view this yesterday, but why not just
`return this.mob.level.getBlockState(this.bHouseBlockPos).getBlock() instanceof BirdhouseBlock`, the if-else statement is redundant here